inginious/frontend/plugins/analytics/api/stacked_bar_plot_api.py

```diff
@@ -0,0 +1,117 @@
+import web
+
+from inginious.frontend.pages.api._api_page import APIError
+from inginious.frontend.plugins.utils.superadmin_utils import SuperadminAPI
+from ..utils import get_api_query_parameters
+
+
+def get_course_ids(data):
+    """" Return a list of course ids """
+    return list(map(lambda course: course["course"], data))
+
+
+def create_y_data_list(num_courses, num_services):
+    """ create zero list to represent y data """
+    return [[0] * num_services for i in range(num_courses)]
```

Comment:
I think this function is not necessary. Just do the same logic in the function where you are calling this. You can add a comment that this creates a list of lists containing zeros.

Reply:
I think the opposite: It's better to have less comments. Normally the functions change, but not the comments, so it's better if the code is understandable without comments.